app/controllers/api/classes_controller.rb

```diff
@@ -88,6 +88,54 @@ def get_single_class
         end
     end
 
+    def update_class
```

Comment:
You can move this entire code into a service like `SkillClassManager::Update(params).execute` `SkillClassManager::Delete(params).execute`
Also, avoid using unnecessary instance variables

Reply:
Other than keeping the code cleaner/shorter in the controller, what's the advantage of having a service with that feature if I'll only call it once?

Reply:
The instance variable is because the class is fetched in the before_action method `check_class` , so the only way I found to access it was through an instante variable

Reply:
* Code readability and reusability. Just because we are not using more than once, doesn't mean we can drop the code anywhere x)
* The other thing is about testing. When you start doing tests you will perceive the major advantage x) usually when testing you mock wtv doesnt matter for exaple:

```ruby
def foo
   params_processed = params
   some_code
   
   SomeNiceService.new.execute
   
   return response
end
```

With the code above you can test the method and everything which runs inside. However, tests should be as specific as possible. So, if you want to test params, some_code and the response it doesnt matter to test the SomeNiceService logic. So you mock `SomeNiceService` to return always true (no errors, and fixed expected behavior)

```ruby
    before do
       allow(SomeNiceService).to receive(:execute).and_return(true)
    end

    it 'does not report a error' do
      expect(Exceptions::Reporter).to_not receive(:error)

      subject #execute something
    end
```

Reply:
btw, this is some more complex approach on tests. Don't feel overwhelmed 🙏 but it's perfectly normal it sounds complex because it is. Still, it gives you visibility how things should be done, even if you can't implement them right now. Eventually you will understand better the benefits as you hit some problems on your code

Reply:
Yap, I have to admit I'm really confused atm 😅
I understand services and modules are a much better way to organize code, I just felt that due to the scale of this project it wouldn't be fundamental